components/omega/src/ocn/Eos.h

```diff
@@ -0,0 +1,257 @@
+#ifndef OMEGA_EOS_H
+#define OMEGA_EOS_H
+//===-- ocn/Eos.h - Equation of State --------------------*- C++ -*-===//
+//
+/// \file
+/// \brief Contains functors for calculating specific volume
+///
+/// This header defines functors to be called by the time-stepping scheme
+/// to calculate the specific volume based on the choice of EOS
+//
+//===----------------------------------------------------------------------===//
+
+#include "AuxiliaryState.h"
+#include "Config.h"
+#include "EosConstants.h"
+#include "HorzMesh.h"
+#include "MachEnv.h"
+#include "OmegaKokkos.h"
+#include "TimeMgr.h"
+#include <string>
+
+namespace OMEGA {
+
+enum class EosType {
+   Linear,       /// Linear equation of state
+   TEOS10Poly75t /// Roquet et al. 2015 75 term expansion
+};
+
+//
+/// TEOS10 75-term polynomial
+class TEOS10Poly75t {
+ public:
+   Array2DReal specVolPcoeffs;
+
+   /// constructor declaration
+   TEOS10Poly75t(int NVertLevels);
+
+   KOKKOS_FUNCTION void operator()(const Array2DReal &SpecVol, I4 ICell,
```

Comment:
I actually find this operator() overload confusing. I would prefer an explicit calcSpecVolLinear and calcSpecVolTEOS (and we'd need to something like that if we eliminate the specific classes as I suggested above).

Reply:
@philipwjones please let me know if the above restructuring to a single class addresses this issue.  Specifically, were you thinking that the change between `calcSpecVolLinear` and `calcSpecVolTEOS` would happen within the EOS class (as I have done with this commit) or the change would happen "out in the code" elsewhere when the EOS is called?

Reply:
@katsmith133 Yes, this implementation is what I was after. Just a single compute interface with the branch inside. Thanks.

Reply:
Great, thanks!